woocommerce/woocommerce-android
# [Coupons] M1 analytics

Fixes #6323 and adds the missing event tracking for coupons.

**To test:**
1. Turn on Flight mode
2. Go to Coupons
3. Wait for a loading failure
4. Verify that `*_coupons_load_failed` is tracked
5. Leave coupons and turn off Flight mode
6. Go back to coupons
7. Verify that `*_coupons_loaded` is tracked
8. Tap on **Search** 
9. Verify that `*_coupons_list_search_tapped` is tracked
10. Go to a coupon detail
11. Verify that `*_coupon_details` with `loaded` action is tracked
12. Tap on the menu and select **Copy coupon code**
13. Verify that `*_coupon_details` with `copied_code` action is tracked
14. Tap on the menu and select **Share coupon**
15. Verify that `*_coupon_details` with `shared_code` action is tracked
16. Tap on the menu and select **Delete coupon**
17. Verify that `*_coupon_details` with `tapped_delete` action is tracked
